### DISCUSSION TITLE
📌 TODO

### DISCUSSION THREAD

**Discussion by daijro** [daijro/camoufox#40] [Fingerprinting]
# TODO List

Here is a list of data that needs to be spoofed in Camoufox, in no particular order. While these aren't typically seen in WAFs, they can potentially be used to fingerprint a user.

- Better fingerprint rotation library (WIP)
- CPU and GPU canvas image rendering
- Audio signal measurement (without creating noise)
- Performance module randomization
- Plugin rotation
- Mime types
- More history config (navigation.index, document.referrer, navigation.type)
- Bundle other common fonts to rotate
- Storage, quota/usage, memory heap size, & recursion limit
- Keyboard locale and layout

**Comment by daijro** [comment]
More information on Canvas fingerprinting added [here](https://github.com/daijro/camoufox/discussions/41).

**Comment by snirow** [comment]
Hi, I've collected some WebGL fingerprints. How can I share it with you? I have it in sqlite at the moment.

**Reply by daijro** [reply]
Hello! You can share them with me through email:
daijro.dev@gmail.com

**Comment by olexayko** [comment]
Can I somewhere save fingerprint data and reuse it than? I need to run each profile with assigned constant fingerprint

**Reply by daijro** [reply]
Hello, Camoufox launch options can be exported like this:

```python
from camoufox.utils import launch_options
# Pass in all Camoufox/AsyncCamoufox parameters here
data = launch_options(headless=False, os='mac', ...)
```

This exports all the launch options that Camoufox passes to Playwright.

---

You can use this config by passing it into the Camoufox or AsyncCamoufox methods:

```python
from camoufox.sync_api import Camoufox
with Camoufox(from_options=data) as browser:
    page = browser.new_page()
```

Or directly into Playwright:
```python
with sync_playwright() as p:
    browser = p.firefox.launch(**data)
    page = browser.new_page()
```

**Comment by olexayko** [comment]
cant i run profile with randomized parameters first, and get launch_options after that?
